### PR TITLE
test: Add unit tests to advanced_cluster

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Add test for refresh functions. 

Link to any related issue(s): https://jira.mongodb.org/browse/CLOUDP-214960

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

**Review by marcosuma — APPROVED** [review]

File: internal/service/advancedcluster/common_advanced_cluster_test.go [hunk 1]
```diff
@@ -30,3 +47,240 @@ func TestRemoveLabel(t *testing.T) {
...
+	args := a.Called(ctx, groupID)
+	var response = args.Get(0).(ClusterResponse)
+	return response.advancedClusterResponse, response.response, response.error
+}
+
+func (a *MockClusterService) GetAdvancedCluster(ctx context.Context, groupID, clusterName string) (*matlas.AdvancedCluster, *matlas.Response, error) {
+	args := a.Called(ctx, groupID)
+	var response = args.Get(0).(ClusterResponse)
+	return response.advancedCluster, response.response, response.error
+}
```
Comment (AgustinBettati) [inline internal/service/advancedcluster/common_advanced_cluster_test.go]: Should we generate the MockService with mockery?

Comment (oarbusi) [reply internal/service/advancedcluster/common_advanced_cluster_test.go]: You are right! Did this tests before we decided to use mockery, and copy pasted them from previous PR. I already changed it.

**Review by maastha — APPROVED** [review]
LGTM

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Pull request merged**